_minutes/2024-11-07-wecg.md

```diff
@@ -0,0 +1,184 @@
+ * [oliver] Treating this as opaque is fine, but I'm curious what this ID looks like in Firefox.
+ * [rob] It's generated in Firefox. I believe it's the timestamp. On click, a weak ref to the element is stored, timestamp is captured and passed to the extension.
+ * [oliver] Thanks, just wanted to understand the general shape.
+ * [timothy] I'd want it to be opaque. UUID or some string.
+ * [rob] It's currently an integer. This enables extensions to pass the value to getTargetElement in a code string (tabs.executeScript) without serialization concerns. That concern is lesser in MV3 because scripting.executeScript does not accept code strings, and accepts args for func instead.
+ * [timothy] Want it to be opaque so there's no meaning associated with it. Allows us to modify the implementation later if necessary. E.g. across processes, etc.
+ * [rob] I don't think that it should be a string; the integer can be a random number, as it is only used to avoid race conditions where “the last clicked element” changed between the time of the invocation of the context menu and the invocation of the script.
+ * [tomislav] I don't think Timothy means a value you can't look into, just a value without meaning.
+ * [timothy] Exactly. Not a timestamp. No inherent meaning to the value.
+ * [rob] Thanks for the clarification - I thought that you meant a string or object by “opaque”. An integer sounds good to me.
+ * [rob] Safari seems on board. Does Chrome have any oppositions?
+ * [oliver] I can follow up.
+
+[Issue 717](https://github.com/w3c/webextensions/issues/717): Request: [ContextMenus] Include `<canvas>`s in the "page" ContextType (or it's own ContextType)
+
+ * [simeon] The only way to match &lt;canvas> elements is to specify “all”. The request is to offer a specific contextType to match &lt;canvas>.
+ * [tomislav] I'm supportive of a canvas context type. No implementation concerns come to mind.
+ * [timothy] Commented on the issue, but I'm also supportive of a canvas type.
+ * [rob] What useful information would an extension get for a canvas? There is no URL or other relevant context.
+ * [timothy] I think this is related to the previous issue where you need to be able to work with the element to meaningfully interact with it. They lean on each other.
+ * [simeon] Sounds like we're aligned?
+ * [rob] I'd like to see more details on use cases. One of the comments was to support a CSS selector to match context menus by selector. That would be a more general way to support this capability. I'm generally in favor of CSS selector. I'm opposed to adding “canvas” ContextType when there is no special context associated with it.
+ * [timothy] We'd still have the issue of needing getTargetElement to meaningfully interact with the element
+ * [simeon] Agreed, I think we're discussing which solution to issue 717 is the right approach.
+ * [timothy] I'm in favor of “canvas” context type because it is simpler than matching by CSS selector.
```

Comment:
```suggestion
 * [timothy] I'm in favor of “canvas” context type because it is simpler than matching by CSS selector. Not worried about API bloat if this case later becomes addressable with selector matching.
```